### ISSUE TITLE
[bug] FeatureExtraction node reports fatal error "Invalid imageDescriber: tag16h5" when using AprilTags

### ISSUE THREAD

**Issue by MaxDidIt** [alicevision/Meshroom#2175]
I wanted to test the AprilTag recognition of Meshroom, so I created a small image set consisting only of two pieces of paper with both `cctag3` and `tag16h5` tags on them, like this:

![20230902_205035_small](https://github.com/alicevision/Meshroom/assets/4189913/de2034ae-db41-4e0f-aad3-db9be5c4a4c6)

I then activated the `cctag3` and `tag16h5` describer types in the FeatureExtraction node:

![grafik](https://github.com/alicevision/Meshroom/assets/4189913/098463d1-9e06-485c-a107-960ef47611f6)

However, when I start the pipeline process in Meshroom, the FeatureExtraction node immediately exits with a fatal error saying `Invalid imageDescriber : tag16h5`.

Here is the complete log of the FeatureExtraction node:

```
[2023-09-02 21:16:54.205103] [0x00002584] [trace]   Embedded OCIO configuration file: 'C:\Photogrammetrie\Meshroom-2023.2.0\aliceVision/share/aliceVision/config.ocio' found.
Program called with the following parameters:
 * contrastFiltering =  Unknown Type "enum aliceVision::feature::EFeatureConstrastFiltering"
 * describerPreset =  Unknown Type "enum aliceVision::feature::EImageDescriberPreset"
 * describerQuality =  Unknown Type "enum aliceVision::feature::EFeatureQuality"
 * describerTypes = "cctag3,dspsift,tag16h5"
 * forceCpuExtraction = 1
 * gridFiltering = 1
 * input = "C:/Photogrammetrie/Markers Test/#2 Photogrammetrie/MeshroomCache/CameraInit/8b27f29f8d2c3603207f311cc2d31818cd30768d/cameraInit.sfm"
 * masksFolder = ""
 * maxCoresAvailable =  Unknown Type "unsigned int" (default)
 * maxMemoryAvailable = 18446744073709551615 (default)
 * maxNbFeatures = 0 (default)
 * maxThreads = 0
 * output = "C:/Photogrammetrie/Markers Test/#2 Photogrammetrie/MeshroomCache/FeatureExtraction/df78b74a7f82c790d615e71dea97cf79608309f1"
 * rangeSize = 40
 * rangeStart = 0
 * relativePeakThreshold = 0.02 (default)
 * verboseLevel = "debug"
 * workingColorSpace =  Unknown Type "enum aliceVision::image::EImageColorSpace"

Hardware : 
	Detected core count : 12
	OpenMP will use 12 cores
	Detected available memory : 25235 Mo

[21:16:54.207103][info] CUDA-Enabled GPU.
Device information:
	- id:                      0
	- name:                    NVIDIA GeForce RTX 2060 SUPER
	- compute capability:      7.5
	- clock frequency (kHz):   1710000
	- total device memory:     8191 MB 
	- device memory available: 7153 MB 
	- per-block shared memory: 49152
	- warp size:               32
	- max threads per block:   1024
	- max threads per SM(X):   1024
	- max block sizes:         {1024,1024,64}
	- max grid sizes:          {2147483647,65535,65535}
	- max 2D array texture:    {131072,65536}
	- max 3D array texture:    {16384,16384,16384}
	- max 2D linear texture:   {131072,65000,2097120}
	- max 2D layered texture:  {32768,32768,2048}
	- number of SM(x)s:        34
	- registers per SM(x):     65536
	- registers per block:     65536
	- concurrent kernels:      yes
	- mapping host memory:     yes
	- unified addressing:      yes
	- texture alignment:       512 byte
	- pitch alignment:         32 byte
CUDA build version: 11.6
0
[21:16:54.375598][fatal] Invalid imageDescriber : tag16h5
```

I'm not sure if I have made a mistake when configuring the FeatureExtraction node or if this is a bug. I would expect the tag16h5 setting to make Meshroom recognize AprilTags the same way the cctag3 setting lets it recognize CCTags.

- I am using Meshroom on Windows 10.
- I am using Meshroom version 2023.2.0 (Downloaded from Meshroom website)
- I am using Python 3.7.4

**Comment by natowi** [comment]
Could you try this with another input path without empty spaces and #?
(/Photogrammetrie/Markers Test/#2 Photogra)

**Comment by MaxDidIt** [comment]
I have changed the input path, but the problem still occurs. See here:

```
[2023-09-03 17:26:33.493174] [0x0000064c] [trace]   Embedded OCIO configuration file: 'C:\Photogrammetrie\Meshroom-2023.2.0\aliceVision/share/aliceVision/config.ocio' found.
Program called with the following parameters:
 * contrastFiltering =  Unknown Type "enum aliceVision::feature::EFeatureConstrastFiltering"
 * describerPreset =  Unknown Type "enum aliceVision::feature::EImageDescriberPreset"
 * describerQuality =  Unknown Type "enum aliceVision::feature::EFeatureQuality"
 * describerTypes = "dspsift,tag16h5,cctag3"
 * forceCpuExtraction = 1
 * gridFiltering = 1
 * input = "C:/Photogrammetrie/markerstest/photogrammetrie/MeshroomCache/CameraInit/f84ddc11729c5fcfc4d41ce776943f7d9bf3cc70/cameraInit.sfm"
 * masksFolder = ""
 * maxCoresAvailable =  Unknown Type "unsigned int" (default)
 * maxMemoryAvailable = 18446744073709551615 (default)
 * maxNbFeatures = 0 (default)
 * maxThreads = 0
 * output = "C:/Photogrammetrie/markerstest/photogrammetrie/MeshroomCache/FeatureExtraction/e6e78f0a53847608b8d85f2817cb0f89af05f530"
 * rangeSize = 40
 * rangeStart = 0
 * relativePeakThreshold = 0.02 (default)
 * verboseLevel = "info"
 * workingColorSpace =  Unknown Type "enum aliceVision::image::EImageColorSpace"

Hardware : 
	Detected core count : 12
	OpenMP will use 12 cores
	Detected available memory : 27154 Mo

[17:26:33.495174][info] CUDA-Enabled GPU.
Device information:
	- id:                      0
	- name:                    NVIDIA GeForce RTX 2060 SUPER
	- compute capability:      7.5
	- clock frequency (kHz):   1710000
	- total device memory:     8191 MB 
	- device memory available: 7153 MB 
	- per-block shared memory: 49152
	- warp size:               32
	- max threads per block:   1024
	- max threads per SM(X):   1024
	- max block sizes:         {1024,1024,64}
	- max grid sizes:          {2147483647,65535,65535}
	- max 2D array texture:    {131072,65536}
	- max 3D array texture:    {16384,16384,16384}
	- max 2D linear texture:   {131072,65000,2097120}
	- max 2D layered texture:  {32768,32768,2048}
	- number of SM(x)s:        34
	- registers per SM(x):     65536
	- registers per block:     65536
	- concurrent kernels:      yes
	- mapping host memory:     yes
	- unified addressing:      yes
	- texture alignment:       512 byte
	- pitch alignment:         32 byte
CUDA build version: 11.6
0
[17:26:33.673673][fatal] Invalid imageDescriber : tag16h5
```

**Comment by natowi** [comment]
Ok, could you try this: one FeatureMatching Node with dspsift and one with Apriltag enabled?

![234966360-2ffcf792-279f-4076-bbc8-1009f611ccee](https://github.com/alicevision/Meshroom/assets/8138537/38bdbc2d-5430-4358-a56f-ac0faf9e8b08)

**Comment by MaxDidIt** [comment]
I tried your setup with two FeatureMatching nodes, but it still fails in the FeatureExtraction node with the same error message:
![grafik](https://github.com/alicevision/Meshroom/assets/4189913/d55dd47f-1f1e-4486-b192-124831702626)

I noticed that in your screenshot, the FeatureExtraction node still passes it's describer types on to the FeatureMatching node, causing it to have the same describers. I also tried a setup where I set the describers in all FeatureExtraction and -Matching nodes manually:
![attempt2](https://github.com/alicevision/Meshroom/assets/4189913/8c0ec50b-f6fd-4698-905c-95a42c7b8c93)

However, this still fails with the same error message.

I also tried to run the first three nodes in parallel completely, but it doesn't seem to have any effect:
![grafik](https://github.com/alicevision/Meshroom/assets/4189913/5476bc66-f1b3-4b30-8dc8-a3e4264c9734)

**Comment by cbentejac** [comment]
Hi @MaxDidIt, do you use Meshroom on Windows or on Linux? 

AprilTags are currently only supported on the Linux version of Meshroom since AprilTag officially only supports Linux operating systems (https://github.com/AprilRobotics/apriltag#install), so that would explain your issue if you're on Windows.

**Comment by MaxDidIt** [comment]
Ah, that seems to be the issue then. I am running Meshroom on Windows currently.

Can you make any prediction on wether there's any way for Meshroom to support AprilTags on Windows at some point or will Linux be the only way to use this feature?

**Comment by cbentejac** [comment]
There is no plan in the near future to support AprilTags on the Windows version of Meshroom. It might happen at some point but it is not a priority at the moment. 

Should Windows become officially supported by AprilTag someday, we will definitely add that support as well since it will not require any tricks and/or workarounds on our part to get it to work.

**Comment by simogasp** [comment]
> There is no plan in the near future to support AprilTags on the Windows version of Meshroom. It might happen at some point but it is not a priority at the moment.
> 
> Should Windows become officially supported by AprilTag someday, we will definitely add that support as well since it will not require any tricks and/or workarounds on our part to get it to work.

What is the problem with AprilTags and Windows? Normally, it should be cross-platform. Is it a shared library issue?
